### ISSUE TITLE
Nexmo::Error

### ISSUE THREAD

**Issue by kazmin** [Vonage/vonage-ruby-sdk#35]
If we don't have status = 0, can we put the status variable in the Error class so we don't have to parse it from the error message when an exception is thrown.

**Comment by timcraft** [comment]
@kazmin I can understand it would be awkward to get the status out of the error message, the error messages aren't meant to be parsed. I think it might be better to remove the status check from #send_message, then that would be more flexible as you could check the status code directly.

Unfortunately that's not backwards compatible behaviour so might have to wait. If that approach fits your immediate needs better you could always monkey patch it in, something like this:

``` ruby
class Nexmo::Client
  def send_message(params)
    post('/sms/json', params)
  end
end
```

**Comment by Sacristan** [comment]
any estimate ?

**Comment by ekyfauzi** [comment]
Hi, please check this
https://github.com/timcraft/nexmo/pull/40

I have similar problem, when message fail to send, i can do anything, it just raise an error.
Well, i try to fix that when message fail to send, it's not rase an error.

**Comment by timcraft** [comment]
@ekyfauzi Planned change for the next major version is to simplify #send_message, as per the above example. Recommended workaround in the meantime is to implement that yourself--options include monkey-patching, subclassing, or adding a new method with a different name.

**Comment by timcraft** [comment]
Implemented in 85f8529, you can now check for the status directly in the response object. No more awkward parsing of exception messages.
